src/libs/actions/connections/NetSuiteCommands.ts

```diff
@@ -0,0 +1,274 @@
+                                pendingFields: {
+                                    [settingName]: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE,
+                                },
+                                errorFields: {
+                                    [settingName]: null,
+                                },
+                            },
+                        },
+                    },
+                },
+            },
+        },
+    ];
+
+    const failureData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                connections: {
+                    netsuite: {
+                        options: {
+                            config: {
+                                [settingName]: oldSettingValue ?? null,
+                                pendingFields: {
+                                    [settingName]: null,
+                                },
+                                errorFields: {
+                                    [settingName]: ErrorUtils.getMicroSecondOnyxErrorWithTranslationKey('common.genericErrorMessage'),
+                                },
+                            },
+                        },
+                    },
+                },
+            },
+        },
+    ];
+
+    const successData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                connections: {
+                    netsuite: {
+                        options: {
+                            config: {
+                                pendingFields: {
+                                    [settingName]: null,
+                                },
+                                errorFields: {
+                                    [settingName]: null,
+                                },
+                            },
+                        },
+                    },
+                },
+            },
+        },
+    ];
+    return {optimisticData, failureData, successData};
+}
+
+function updateNetSuiteExporter(policyID: string, tokenSecret: string, exporter: string, oldExporter: string) {
+    const onyxData = updateNetSuiteOnyxData(policyID, CONST.NETSUITE_CONFIG.EXPORTER, exporter, oldExporter);
+
+    const parameters = {
+        policyID,
+        email: exporter,
+        authToken: tokenSecret,
+    };
+    API.write(WRITE_COMMANDS.UPDATE_NETSUITE_EXPORTER, parameters, onyxData);
+}
+
+function updateNetSuiteExportDate(policyID: string, tokenSecret: string, date: ValueOf<typeof CONST.NETSUITE_EXPORT_DATE>, oldDate?: ValueOf<typeof CONST.NETSUITE_EXPORT_DATE>) {
+    const onyxData = updateNetSuiteOnyxData(policyID, CONST.NETSUITE_CONFIG.EXPORT_DATE, date, oldDate);
+
+    const parameters = {
+        policyID,
+        value: date,
+        authToken: tokenSecret,
```

Comment:
Why do we need to pass authToken?

Reply:
yeah it got me confused with the `authToken` of the existing API call. Is it `tokenSecret` from NetSuite? I wasn't aware it was needed. I don't think we should be storing the `tokenSecret` in the Onyx or should be needing to pass in the API calls. That looks like a security risk @yuwenmemon.


Quick clarification, is this what you meant when you asked in my PR to add `authToken`? My API for updateNetSuiteSubsidiary worked without passing the tokenSecret.

Reply:
The `authToken` mentioned in the doc is the authToken that is passed in by default in every API call. We don't need to pass in the tokenSecret for every single NetSuite API call. The back-end handles managing tokens for that. We can just remove this line and every single one like it.